adafruit_esp32spi/adafruit_esp32spi.py

```diff
@@ -462,17 +491,57 @@ def connect_AP(self, ssid, password): # pylint: disable=invalid-name
             self.wifi_set_passphrase(ssid, password)
         else:
             self.wifi_set_network(ssid)
-        for _ in range(10): # retries
+        times = time.monotonic()
+        while (time.monotonic() - times) < timeout_s:  # wait up until timeout
```

Comment:
switched `connect_AP` to also be timeout based instead of retry count based. Could speed up perceived connection time since we're checking more frequently now. (every 0.05 seconds as opposed to every 1 second)

Reply:
This seems fine, may be useful as you say. How did you come to 0.05 seconds as a sleep time?

Reply:
I came up with the value to match this: https://github.com/adafruit/Adafruit_CircuitPython_ESP32SPI/blob/master/adafruit_esp32spi/adafruit_esp32spi.py#L175-L180

It seemed like a reasonable sleep time. I saw in other places 0.01 is used, but went with the slightly longer one.

I don't really have a preference for what time we use.

Reply:
This is fine